sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1502,6 +1521,20 @@ public TableReference getTable() {
       return toBuilder().setQueryLocation(location).build();
     }
 
+    /**
+     * Temporary dataset reference when using {@link #fromQuery(String)}. When reading from a query,
+     * BigQuery will create a temporary dataset and a temporary table to store the results of the
+     * query. With this option, you can set an existing dataset to create the temporary table.
+     * BigQueryIO will create a temporary table in that dataset, and will remove it once it is not
+     * needed. No other tables in the dataset will be modified. If your job does not have
+     * permissions to create a new dataset, and you want to use {@link #fromQuery(String)} (for
+     * instance, to read from a view), you should use this option. Remember that the dataset must
+     * exist and your job needs permissions to create and remove tables inside that dataset.
```

Comment:
We should also make sure that any table that Beam create or delete dynamically does not conflict with an existing table in the Dataset (at runtime).

Reply:
I have just added two commits. If the user specifies the temp dataset and it is using `fromQuery`
* check that the specified dataset exists
* check that the destination table does not exist, to avoid overwriting any existing table in the dataset specified by the user (unlikely, due to the random generation of uuids for the temp tables, but not impossible)